src/components/QueryEditor/utils/sql.utils.ts

```diff
@@ -7,12 +7,23 @@ import {
   QueryEditorPropertyExpression,
   QueryEditorPropertyType,
 } from '../expressions';
-import { SQLQuery, SQLExpression } from '../types';
+import { SQLQuery, SQLExpression, DB } from '../types';
 
-export function defaultToRawSql({ sql, dataset, table }: SQLQuery): string {
+export function getRawSqlFN(db: DB) {
+  if (db.toRawSql) {
+    return db.toRawSql;
+  }
+
+  if (db.disableDatasets) {
+    return toRawSqlWithoutDataset;
```

Comment:
hmm.. these two functions are super-similar, they only differ in 3 lines. i wonder if we could do something like... 
pseudocode
```
function toRawSQL({sql, dataset, table}:SQLQuery, hasDataset:boolean) {
.
.
.
// here is the IF based on `hasDataset`

}
```
then in getRawSQLFn:

```
export function getRawSqlFn(db:DB) {
.
.
return (q : SQLQuery) => toRawSql(q, true);
.
.
return (q: SQLQuery) => toRawSQL(q, false);
}
```

this would allow you to only have one copy of `toRawSQL`.. though it complicates the other part a bit. WDYT?